valhalla/meili/match_result.h

```diff
@@ -66,6 +66,12 @@ struct MatchResult {
   }
 };
 
+struct MapMatchResult {
+  float confidence_score = 0.f;
+  float raw_score = 0.f;
+  std::vector<MatchResult> match_results;
+};
+
```

Comment:
The most important change is replacing tuple with a new struct `MapMatchResult`. Warnings as errors pointed out that a few constants(i.e `kRawScoreIndex, kRawScoreIndex, kMatchResultsIndex) were unused. + These constants a declared in multiple files. So replacing tuple with a struct with named members looks like a good way to go

Reply:
Q: What's the right file/module to declare this struct? I think of moving it to the thor module but not sure what's the right file for this purpose

Reply:
yeah its a bit of a mess to be honest. in the first implementation where thor exposed things from meili to the outward facing api there was a lot of duplication of objects. recently we reduced some of that but there is still this...

i think its a bit confusing to name this object like this because we already have these: https://github.com/valhalla/valhalla/blob/master/valhalla/meili/match_result.h#L138-L141

maybe we just add the addtional score to that existing matchresult object and use that?